src/index.js

```diff
@@ -104,22 +108,55 @@ HTTPSnippet.prototype.prepare = function (request) {
       if (request.postData.params) {
         var form = new MultiPartForm()
 
+        // The `form-data` module returns one of two things: a native FormData object, or its own polyfill. Since the
+        // polyfill does not support the full API of the native FormData object, when this library is running in a
+        // browser environment it'll fail on two things:
+        //
+        //  - The API for `form.append()` has three arguments and the third should only be present when the second is a
+        //    Blob or USVString.
+        //  - `FormData.pipe()` isn't a function.
+        //
+        // Since the native FormData object is iterable, we easily detect what version of `form-data` we're working
+        // with here to allow `multipart/form-data` requests to be compiled under both browser and Node environments.
+        //
+        // This hack is pretty awful but it's the only way we can use this library in the browser as if we code this
+        // against just the native FormData object, we can't polyfill that back into Node because Blob and File objects,
+        // which something like `formdata-polyfill` requires, don't exist there.
+        const isNativeFormData = (typeof form[Symbol.iterator] === 'function')
+
         // easter egg
-        form._boundary = '---011000010111000001101001'
+        const boundary = '---011000010111000001101001'
+        if (!isNativeFormData) {
+          form._boundary = boundary
+        }
 
         request.postData.params.forEach(function (param) {
-          form.append(param.name, param.value || '', {
-            filename: param.fileName || null,
-            contentType: param.contentType || null
-          })
+          if (isNativeFormData) {
+            if (isBlob(param.value)) {
+              form.append(param.name, param.value || '', param.fileName || null)
+            } else {
+              form.append(param.name, param.value || '')
+            }
+          } else {
+            form.append(param.name, param.value || '', {
+              filename: param.fileName || null,
+              contentType: param.contentType || null
+            })
+          }
```

Comment:
Since there are a number of times we default `value` and `filename` from `param`, I suggest defaulting once at the beginning of the block and using that result to make sure the item appended in each condition is the same.

```suggestion
          const name = param.name;
          const value = param.value || '';
          const filename = param.fileName || null;
          const contentType = param.contentType || null;
          
          if (isNativeFormData) {
            if (isBlob(value)) {
              form.append(name, value, filename)
            } else {
              form.append(name, value)
            }
          } else {
            form.append(name, value, { filename, contentType });
          }
```

This way the slight variations in each of the three branches are more transparent. 😄